### ISSUE TITLE
CALICO_IPV4POOL_CIDR accepts IPv6 addresses

### ISSUE THREAD

**Issue by lwr20** [projectcalico/calicoctl#1539]
I run calico-node with:
```
$ docker exec -it host sh -c 'docker run --net=host --privileged --name=calico-node -d --restart=always -e CALICO_IPV4POOL_CIDR=fd00::/24 -e ETCD_AUTHORITY=10.240.0.12:2379 -e ETCD_SCHEME=http -e IP6_AUTODETECT_METHOD=first-found -e CALICO_LIBNETWORK_CREATE_PROFILES=true -e CALICO_LIBNETWORK_LABEL_ENDPOINTS=false -e NODENAME=2b6048a03e65 -e CALICO_NETWORKING_BACKEND=bird -e NO_DEFAULT_POOLS= -e CALICO_LIBNETWORK_ENABLED=true -e IP_AUTODETECT_METHOD=first-found -e CALICO_LIBNETWORK_IFPREFIX=cali -e ETCD_ENDPOINTS= -v /var/log/calico:/var/log/calico -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock quay.io/calico/node:v1.1.0-rc5'
```
expecting it to fail, and it does not:
```
$ docker exec -it host sh -c 'docker logs calico-node'
  # time="2017-02-21T19:53:16Z" level=info msg="Loading config from environment"
  # Skipping datastore connection test
  # time="2017-02-21T19:53:16Z" level=info msg="Building new node resource" Name=2b6048a03e65
  # time="2017-02-21T19:53:16Z" level=info msg="Initialise BGP data"
  # Using autodetected IPv4 address on interface eth0: 172.17.0.4/16
  # No AS number configured on node resource, using global value
  # time="2017-02-21T19:53:16Z" level=info msg="Ensure default IPv6 pool is created. IPIP enabled: false"
  # Created default IPv6 pool (fd00::/24) with NAT outgoing enabled. IPIP enabled: false
  # time="2017-02-21T19:53:16Z" level=info msg="IPv6 supported on this platform: true"
  # time="2017-02-21T19:53:16Z" level=info msg="Ensure default IPv6 pool is created. IPIP enabled: false"
  # Created default IPv6 pool (fd80:24e2:f998:72d6::/64) with NAT outgoing enabled. IPIP enabled: false
  # time="2017-02-21T19:53:16Z" level=info msg="Error enumerating host directories" error="100: Key not found (/calico/v1/host) [29]"
  # time="2017-02-21T19:53:16Z" level=info msg="Ready flag is now set"
  # time="2017-02-21T19:53:16Z" level=info msg="Assigned cluster GUID" ClusterGUID=f7d8f643e959410481ee132858608bb4
  # Using node name: 2b6048a03e65
  # time="2017-02-21T19:53:16Z" level=info msg="Loading config from environment"
  # Starting libnetwork service
  # Calico node started successfully
```
It seems to have created 2 default IPv6 pools in this case...

**Comment by lwr20** [comment]
Seems that the reverse is also true - `CALICO_IPV6POOL_CIDR` accepts IPv4 addresses, and creates 2 default pools:
```
$ docker exec -it host sh -c 'docker run --net=host --privileged --name=calico-node -d --restart=always -e CALICO_IPV6POOL_CIDR=1.2.3.4/24 -e IP_AUTODETECT_METHOD=first-found -e CALICO_LIBNETWORK_CREATE_PROFILES=true -e CALICO_LIBNETWORK_LABEL_ENDPOINTS=false -e CALICO_LIBNETWORK_IFPREFIX=cali -e ETCD_ENDPOINTS= -e NODENAME=2b6048a03e65 -e CALICO_NETWORKING_BACKEND=bird -e NO_DEFAULT_POOLS= -e CALICO_LIBNETWORK_ENABLED=true -e IP6_AUTODETECT_METHOD=first-found -e ETCD_AUTHORITY=10.240.0.12:2379 -e ETCD_SCHEME=http -v /var/log/calico:/var/log/calico -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock quay.io/calico/node:v1.1.0-rc5'
```
```
$ docker exec -it host sh -c 'docker logs calico-node'
  # time="2017-02-21T19:53:08Z" level=info msg="Loading config from environment"
  # Skipping datastore connection test
  # time="2017-02-21T19:53:08Z" level=info msg="Building new node resource" Name=2b6048a03e65
  # time="2017-02-21T19:53:08Z" level=info msg="Initialise BGP data"
  # Using autodetected IPv4 address on interface eth0: 172.17.0.4/16
  # No AS number configured on node resource, using global value
  # time="2017-02-21T19:53:08Z" level=info msg="Ensure default IPv4 pool is created. IPIP enabled: false"
  # Created default IPv4 pool (192.168.0.0/16) with NAT outgoing enabled. IPIP enabled: false
  # time="2017-02-21T19:53:08Z" level=info msg="IPv6 supported on this platform: true"
  # time="2017-02-21T19:53:08Z" level=info msg="Ensure default IPv4 pool is created. IPIP enabled: false"
  # Created default IPv4 pool (1.2.3.0/24) with NAT outgoing enabled. IPIP enabled: false
  # time="2017-02-21T19:53:08Z" level=info msg="Error enumerating host directories" error="100: Key not found (/calico/v1/host) [8]"
  # time="2017-02-21T19:53:08Z" level=info msg="Ready flag is now set"
  # time="2017-02-21T19:53:08Z" level=info msg="Assigned cluster GUID" ClusterGUID=c8721bc3c7a84bb788537e62f57cc95f
  # Using node name: 2b6048a03e65
  # time="2017-02-21T19:53:08Z" level=info msg="Loading config from environment"
  # Starting libnetwork service
  # Calico node started successfully
```

**Comment by tmjd** [comment]
I believe this is fixed via #1542, should I close this issue and the others like it since that is merged?

**Comment by lwr20** [comment]
Fix looks good in calico/node:v1.1.0-rc5-14-g1f5381f3
